components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -74,26 +82,202 @@
     }
 
     @Override
     public AuthenticatorFlowStatus process(HttpServletRequest request,
                                            HttpServletResponse response, AuthenticationContext context)
             throws AuthenticationFailedException, LogoutFailedException {
 
+        Cookie autoLoginCookie = getAutoLoginCookie(request.getCookies());
+
         if (context.isLogoutRequest()) {
             return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
-        } else {
-            return super.process(request, response, context);
+        } else if (autoLoginCookie != null && isEnableAutoLoginAfterPasswordReset(context)) {
+            try {
+                return executeAutoLoginFlow(request, response, context, autoLoginCookie);
+            } catch (AuthenticationFailedException e) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Error occurred while executing the Auto Login from Cookie flow: " + e);
+                }
+                removeAutoLoginCookie(response, autoLoginCookie);
+            }
+        }
+        return super.process(request, response, context);
+    }
+
+    private AuthenticatorFlowStatus executeAutoLoginFlow(HttpServletRequest request, HttpServletResponse response,
+                                                         AuthenticationContext context, Cookie autoLoginCookie)
+            throws AuthenticationFailedException {
+
+        JSONObject cookieValueJSON = transformCookieValueToJSON(autoLoginCookie);
+        String usernameInCookie = (String) cookieValueJSON.get(USERNAME);
+        String usernameInHttpRequest = request.getParameter(USERNAME);
+
+        if (log.isDebugEnabled()) {
+            log.debug("Started executing Auto Login from Cookie flow.");
+        }
+
+        if (StringUtils.isNotEmpty(usernameInHttpRequest) && StringUtils.isNotEmpty(usernameInCookie) &&
+                !StringUtils.equalsIgnoreCase(usernameInHttpRequest, usernameInCookie)) {
+            throw new AuthenticationFailedException("Username in HTTP Request: " + usernameInHttpRequest
+                    + " and username in Cookie: " + usernameInCookie + " does not match.");
+        }
+
+        validateCookieSignature(usernameInCookie, cookieValueJSON);
+        usernameInCookie = FrameworkUtils.prependUserStoreDomainToName(usernameInCookie);
+
+        String tenantDomain = MultitenantUtils.getTenantDomain(usernameInCookie);
+        UserStoreManager userStoreManager = getUserStoreManager(usernameInCookie);
+
+        usernameInCookie = getMultiAttributeUsername(usernameInCookie, tenantDomain, userStoreManager);
+        context.setSubject(AuthenticatedUser.createLocalAuthenticatedUserFromSubjectIdentifier(usernameInCookie));
+        removeAutoLoginCookie(response, autoLoginCookie);
+        return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+    }
+
+    private void removeAutoLoginCookie(HttpServletResponse response, Cookie autoLoginCookie) {
```

Comment:
let's move these private methods to the end of the file, It will be better to have all the public or protected method in top of the class, then the private methods for better code readability

Reply:
Resolved in https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/74/commits/056eab067512701b37497b40fc62d2e23f6a9db9